lightning/impl/tests/test_dataset.py

```diff
@@ -10,26 +12,31 @@
 from lightning.impl.dataset_fast import CSRDataset
 from lightning.impl.dataset_fast import CSCDataset
 
-# Create test datasets.
-X, _ = make_classification(n_samples=20, n_features=100,
-                           n_informative=5, n_classes=2, random_state=0)
-X2, _ = make_classification(n_samples=10, n_features=100,
-                            n_informative=5, n_classes=2, random_state=0)
 
-# Sparsify datasets.
-X[X < 0.3] = 0
+@pytest.fixture(scope="module")
+def test_data():
+    X, _ = make_classification(n_samples=20, n_features=100,
+                               n_informative=5, n_classes=2, random_state=0)
+    X2, _ = make_classification(n_samples=10, n_features=100,
+                                n_informative=5, n_classes=2, random_state=0)
+
+    # Sparsify datasets.
+    X[X < 0.3] = 0
 
-X_csr = sp.csr_matrix(X)
-X_csc = sp.csc_matrix(X)
+    X_csr = sp.csr_matrix(X)
+    X_csc = sp.csc_matrix(X)
 
-rs = check_random_state(0)
-cds = ContiguousDataset(X)
-fds = FortranDataset(np.asfortranarray(X))
-csr_ds = CSRDataset(X_csr)
-csc_ds = CSCDataset(X_csc)
+    rs = check_random_state(0)
+    cds = ContiguousDataset(X)
+    fds = FortranDataset(np.asfortranarray(X))
+    csr_ds = CSRDataset(X_csr)
+    csc_ds = CSCDataset(X_csc)
 
+    return X, X_csr, X_csc, cds, fds, csr_ds, csc_ds
 
-def test_contiguous_get_row():
+
+def test_contiguous_get_row(test_data):
+    X, _, _, cds, _, _, _ = test_data
```

Comment:
This is not support readable... Maybe `test_data` could be a dictionary instead?

Reply:
Yes exactly, just to avoid having to use many underscores.

Reply:
Thank you for your review! Addressed in 148635eb40c5d9f918ceb439cc303d68bdc8f923.